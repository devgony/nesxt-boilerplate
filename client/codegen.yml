overwrite: true
schema: "http://localhost:4000/graphql"
documents: ["./pages/**/*.{js,jsx,ts,tsx}", "./components/**/*.{js,jsx,ts,tsx}"]
generates:
  generated/graphql.tsx:
    plugins:
      - "typescript"
      - "typescript-operations"
      - "typescript-react-apollo"
