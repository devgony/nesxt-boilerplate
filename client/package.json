{
  "name": "client",
  "private": true,
  "scripts": {
    "dev": "npm run gen & next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "gen": "graphql-codegen --config codegen.yml --watch"
  },
  "dependencies": {
    "@apollo/client": "^3.7.15",
    "@graphql-codegen/cli": "^4.0.0",
    "autoprefixer": "^10.4.14",
    "graphql": "^16.6.0",
    "next": "13.4.4",
    "postcss": "^8.4.24",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.44.2",
    "tailwindcss": "^3.3.2"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "4.0.0",
    "@graphql-codegen/introspection": "4.0.0",
    "@graphql-codegen/typescript": "4.0.0",
    "@graphql-codegen/typescript-operations": "4.0.0",
    "@graphql-codegen/typescript-react-apollo": "3.3.7",
    "@types/node": "20.2.5",
    "@types/react": "18.2.7",
    "eslint": "8.41.0",
    "eslint-config-next": "13.4.4",
    "typescript": "5.0.4"
  }
}
