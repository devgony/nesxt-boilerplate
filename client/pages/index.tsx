import { gql, useMutation, useQuery } from "@apollo/client";
import { SubmitHandler, useForm } from "react-hook-form";
import {
  LoginInput,
  LoginMutation,
  LoginMutationVariables,
} from "../generated/graphql";
import { Helmet } from "react-helmet";

const LOGIN = gql`
  mutation login($input: LoginInput!) {
    login(input: $input) {
      ok
      token
      error
    }
  }
`;

const Home = () => {
  const { register, getValues, handleSubmit, formState } = useForm<LoginInput>({
    mode: "onChange",
  });
  const onCompleted = (data: LoginMutation) => {
    const {
      login: { ok, token, error },
    } = data;
    if (ok && token) {
      alert("login success");
    } else {
      alert(error);
    }
  };
  const [login, { data: loginResult, loading }] = useMutation<
    LoginMutation,
    LoginMutationVariables
  >(LOGIN, { onCompleted });

  const onSubmit = ({ name, password }: LoginInput) => {
    login({
      variables: {
        input: { name, password },
      },
    });
  };

  return (
    <div className="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8">
      <Helmet>
        <title>Home | Project Name</title>
      </Helmet>
      <div className="sm:mx-auto sm:w-full sm:max-w-sm">
        <img
          className="mx-auto h-10 w-auto"
          src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
          alt="Your Company"
        />
        <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">
          Sign in to your account
        </h2>
      </div>

      <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
        <form className="space-y-6" onSubmit={handleSubmit(onSubmit)}>
          <div>
            <label
              htmlFor="name"
              className="block text-sm font-medium leading-6 text-gray-900"
            >
              Account name
            </label>
            <div className="mt-2">
              <input
                {...register("name")}
                type="text"
                required
                className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
              />
            </div>
          </div>

          <div>
            <div className="flex items-center justify-between">
              <label
                htmlFor="password"
                className="block text-sm font-medium leading-6 text-gray-900"
              >
                Password
              </label>
              <div className="text-sm">
                <a
                  href="#"
                  className="font-semibold text-indigo-600 hover:text-indigo-500"
                >
                  Forgot password?
                </a>
              </div>
            </div>
            <div className="mt-2">
              <input
                {...register("password")}
                type="password"
                autoComplete="current-password"
                required
                className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
              />
            </div>
          </div>

          <div>
            <button
              type="submit"
              className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            >
              Sign in
            </button>
          </div>
        </form>

        <p className="mt-10 text-center text-sm text-gray-500">
          Not a member?{" "}
          <a
            href="#"
            className="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
          >
            Start a 14 day free trial
          </a>
        </p>
      </div>
    </div>
  );
};

export default Home;
