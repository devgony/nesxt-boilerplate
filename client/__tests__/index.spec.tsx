import { RenderResult, render, screen, waitFor } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import { createMockClient, MockApolloClient } from "mock-apollo-client";
import Home from "@/pages";

describe("<Home />", () => {
  let renderResult: RenderResult;
  let mockedClient: MockApolloClient;
  beforeEach(async () => {
    await waitFor(() => {
      mockedClient = createMockClient();
      renderResult = render(
        <ApolloProvider client={mockedClient}>
          <Home />
        </ApolloProvider>
      );
    });
  });
  it("should render OK", async () => {
    await waitFor(() => {
      expect(document.title).toBe("Home | Project Name");
      expect(renderResult).toMatchSnapshot();
    });
  });
});
