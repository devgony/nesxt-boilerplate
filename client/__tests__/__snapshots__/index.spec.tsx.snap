// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render OK 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8"
      >
        <div
          class="sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <img
            alt="Your Company"
            class="mx-auto h-10 w-auto"
            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
          />
          <h2
            class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
          >
            Sign in to your account
          </h2>
        </div>
        <div
          class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <form
            class="space-y-6"
          >
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="name"
              >
                Account name
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  name="name"
                  required=""
                  type="text"
                />
              </div>
            </div>
            <div>
              <div
                class="flex items-center justify-between"
              >
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="text-sm"
                >
                  <a
                    class="font-semibold text-indigo-600 hover:text-indigo-500"
                    href="#"
                  >
                    Forgot password?
                  </a>
                </div>
              </div>
              <div
                class="mt-2"
              >
                <input
                  autocomplete="current-password"
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  name="password"
                  required=""
                  type="password"
                />
              </div>
            </div>
            <div>
              <button
                class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                type="submit"
              >
                Sign in
              </button>
            </div>
          </form>
          <p
            class="mt-10 text-center text-sm text-gray-500"
          >
            Not a member?
             
            <a
              class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
              href="#"
            >
              Start a 14 day free trial
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8"
    >
      <div
        class="sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <img
          alt="Your Company"
          class="mx-auto h-10 w-auto"
          src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
        />
        <h2
          class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
        >
          Sign in to your account
        </h2>
      </div>
      <div
        class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <form
          class="space-y-6"
        >
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="name"
            >
              Account name
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                name="name"
                required=""
                type="text"
              />
            </div>
          </div>
          <div>
            <div
              class="flex items-center justify-between"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="password"
              >
                Password
              </label>
              <div
                class="text-sm"
              >
                <a
                  class="font-semibold text-indigo-600 hover:text-indigo-500"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
            </div>
            <div
              class="mt-2"
            >
              <input
                autocomplete="current-password"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                name="password"
                required=""
                type="password"
              />
            </div>
          </div>
          <div>
            <button
              class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              type="submit"
            >
              Sign in
            </button>
          </div>
        </form>
        <p
          class="mt-10 text-center text-sm text-gray-500"
        >
          Not a member?
           
          <a
            class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
            href="#"
          >
            Start a 14 day free trial
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
